src/crimson/CMakeLists.txt

```diff
@@ -1,6 +1,13 @@
 add_library(crimson::cflags INTERFACE IMPORTED)
+set(crimson_cflag_definitions "WITH_SEASTAR=1")
+# disable concepts to address https://github.com/boostorg/asio/issues/312
+if((CMAKE_CXX_COMPILER_ID STREQUAL GNU AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 10) OR
+   (CMAKE_CXX_COMPILER_ID STREQUAL Clang))
+ list(APPEND crimson_cflag_definitions
+   "BOOST_ASIO_DISABLE_CONCEPTS")
```

Comment:
OK, we don't plan to use ASIO, so I'm not afraid about disabling concepts there (and forgetting to re-enable when the bug gets fixed).

Reply:
it's `Graylog.h` which pulls boost::asio headers in. i tried to avoid using the all-in-one convenient header of `boost/asio.hpp` , but no luck:

```
In file included from /opt/ceph/include/boost/asio/basic_socket.hpp:19,
                 from /opt/ceph/include/boost/asio/basic_datagram_socket.hpp:20,
                 from /opt/ceph/include/boost/asio/ip/udp.hpp:19,
                 from ../src/common/Graylog.h:7,
                 from ../src/common/ceph_context.cc:36:
/opt/ceph/include/boost/asio/async_result.hpp:70:20: error: a variable concept cannot be constrained
   70 | BOOST_ASIO_CONCEPT completion_handler_for =
      |                    ^~~~~~~~~~~~~~~~~~~~~~
/opt/ceph/include/boost/asio/async_result.hpp:492:20: error: a variable concept cannot be constrained
  492 | BOOST_ASIO_CONCEPT completion_token_for = requires(T&& t)
      |                    ^~~~~~~~~~~~~~~~~~~~
```